service/controller/clusterapi/v29/cluster_resource_set.go

```diff
@@ -531,14 +531,14 @@ func NewClusterResourceSet(config ClusterResourceSetConfig) (*controller.Resourc
 	resources := []controller.Resource{
 		awsClientResource,
 		machineDeploymentResource,
-		clusterAZsResource,
 		accountIDResource,
 		natGatewayAddressesResource,
 		peerRoleARNResource,
 		routeTableResource,
 		vpcCIDRResource,
 		tccpOutputsResource,
 		tccpSubnetResource,
+		clusterAZsResource,
```

Comment:
Availability Zone inventory must be done only after TCCP subnets have been collected.

Reply:
Have you checked if this ordering causes any unexpected problems? Is there a way to verify?

Reply:
I haven't given this PR a test run yet, but given I added this resource last week and it should be only used in `tccpResource` below, I'm optimistic here. Reason why I moved it here is that previous resource `tccpSubnetResource` is the one which fetches that `[]*ec2.Subnet` slice that this uses. Nevertheless, running this operator & creating a cluster would reveal problems.